onosproject/onos-test
# Add onos-gui, envoy setup interface

This PR adds 

-  onos-gui setup interface that users can deploy or enable it like other subsystems. 
-  envoy setup interface that users can deploy or enable envoy proxies. This still needs more work but it works fine for onos-gui. 
-  Adds preliminary functions for creating k8s config maps and ingress resources. This part needs more work to make them more generic. 

To test :
`onit create cluster --set onos-gui.enabled=true --set onos-gui.replicas=1`

and then check k8s pods as follows:
```
onos-test git:(add_gui) kubectl get pods -n onos -w
NAME                                 READY   STATUS    RESTARTS   AGE
atomix-controller-756f86cc9c-m5bz7   1/1     Running   0          87s
database-1-0                         1/1     Running   0          84s
onos-config-7f77c68c88-8chqx         1/1     Running   0          84s
onos-config-envoy-6788bf79d7-zv4gg   1/1     Running   0          84s
onos-gui-74bfff7b99-rdc4x            1/1     Running   0          67s
onos-topo-5bdc4f568-q5cn2            1/1     Running   0          84s
onos-topo-envoy-5c884fc76c-26jnt     1/1     Running   0          76s
```

P.S. This PR is based on the orginal version of onos-gui which uses a separate envoy service for each onos subsystems (e.g. onos-config-envoy, onos-topo-envoy) but it can be changed easily to deploy one envoy service (e.g. onos-envoy) which serves both of onos-config and onos-topo. subsystems.
